### PR TITLE
HtmlMesh Updates

### PR DESCRIPTION
**Pull request by ericwood73**
- Fixed issues when canvas is offset or repositioned which was preventing the HtmlMesh from working in PG if the editor was open
- Add a dispose method to the renderer to allow the DOM elements and listeners as well as the scene observables to be removed when the scene is destroyed (e.g. PG is executed)
- Handle rotations about multiple axes
- Capture pointer events while dragging to allow a camera rotation that brings the pointer over the mesh to trigger capture
- Address some bugs with pointer capture causing inconsistent capturing behavior when a skybox or background is included.

### CODE REVIEW

File: HtmlMesh/src/pointer-events-capture-behavior.ts [hunk 1]
```diff
@@ -37,73 +32,92 @@ const startCaptureOnEnter = (scene: Scene) => {
     if (captureOnEnterCount === 0) {
         document.addEventListener("pointermove", onPointerMove);
         _scene = _scene ?? scene;
-        let rect = _scene
-            .getEngine()
-            .getRenderingCanvasClientRect() as ClientRect;
-        _canvasRect =
-            _canvasRect ??
-            new DOMRect(rect.left, rect.top, rect.width, rect.height);
+        console.log(
```
Comment (sebavan) [inline HtmlMesh/src/pointer-events-capture-behavior.ts:35]: Can you use Logger.log from the Babylon logger instead ?

File: HtmlMesh/src/pointer-events-capture-behavior.ts [hunk 2]
```diff
@@ -37,73 +32,92 @@ const startCaptureOnEnter = (scene: Scene) => {
...
+        _scene.onDisposeObservable.add(doStopCaptureOnEnter);
     }
     captureOnEnterCount++;
 };
 
+const doStopCaptureOnEnter = () => {
+    document.removeEventListener("pointermove", onPointerMove);
+    _scene = null;
+    console.log(
+        "PointerEventsCaptureBehavior: Stopping observation of pointer move events."
```
Comment (sebavan) [inline HtmlMesh/src/pointer-events-capture-behavior.ts:47]: Can you use Logger.log from the Babylon logger instead ?

File: HtmlMesh/src/util.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            lastCallTime = currentTime;
+            // Return the canvas rect, or the last known value
+            let newCanvasRect = scene
+                ?.getEngine()
+                .getRenderingCanvasClientRect();
+            if (newCanvasRect) {
+                canvasRect = newCanvasRect;
+            } else {
+                if (canvasRect) {
+                    console.warn(
```
Comment (sebavan) [inline HtmlMesh/src/util.ts:28]: Same :-) Logger.warn instead ?

File: HtmlMesh/src/util.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                        "Canvas rect became null.  Returning last known value"
+                    );
+                }
+                console.warn("Failed to get canvas rect.");
```
Comment (sebavan) [inline HtmlMesh/src/util.ts:32]: Same :-) Logger.warn instead ?

File: HtmlMesh/src/pointer-events-capture-behavior.ts (hunk 1 above)
Comment (ericwood73) [reply HtmlMesh/src/pointer-events-capture-behavior.ts]: Done

File: HtmlMesh/src/pointer-events-capture-behavior.ts (hunk 2 above)
Comment (ericwood73) [reply HtmlMesh/src/pointer-events-capture-behavior.ts:48]: Done

File: HtmlMesh/src/util.ts (hunk 3 above)
Comment (ericwood73) [reply HtmlMesh/src/util.ts]: Done

File: HtmlMesh/src/util.ts (hunk 4 above)
Comment (ericwood73) [reply HtmlMesh/src/util.ts]: Done

**Review by sebavan — APPROVED** [review]

**Pull request merged**